contrib/IECoreArnold/src/IECoreArnold/procedural/Procedural.cpp

```diff
@@ -99,7 +99,16 @@ static void initialisePython()
 	PyEval_ReleaseThread( PyThreadState_Get() );
 }
 
-static int procInit( AtNode *node, void **userPtr )
+AI_PROCEDURAL_NODE_EXPORT_METHODS(IECoreProcedural);
+
+node_parameters
+{
+	AiParameterStr( "className", "" );
+	AiParameterInt( "classVersion", 1 );
+	AiParameterArray( "parameterValues", AiArray(0, 1, AI_TYPE_STRING) );
+}
+
+procedural_init
```

Comment:
For what it's worth, I hate the way Arnold uses macros to obfuscate the function parameters, and I explicitly wrote out the signatures instead so they were obvious.

Reply:
OK, though I liked having less typing ...